lib/api.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * API Endpoints.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+/**
+ * Validates that a preference name is valid, as part of a REST API request.
+ *
+ * @param  string           $preference_name Preference name.
+ * @param  WP_REST_Request  $request The REST request.
+ * @param  string           $request The parameter key for the value being validated.
+ * @return bool             If the preference name is valid.
+ */
+function gutenberg_validate_preference_name( $preference_name, $request, $key ) {
```

Comment:
We don't _need_ to, but merging these and the Gutenberg_User_Preferences would couple it to the API, as it would become dependent on the API request. But perhaps that's not a problem, I'm still getting used to our coding styles and usually I'd have a class that handled the preferences and could be reused outside of the API. I'll change this round.